app/views/courses/TeacherClassesView.coffee

```diff
@@ -97,10 +97,14 @@ module.exports = class TeacherClassesView extends RootView
     @supermodel.trackCollection(@classrooms)
     @listenTo @classrooms, 'sync', ->
       for classroom in @classrooms.models
+        continue if classroom.get('archived')
         classroom.sessions = new LevelSessions()
-        jqxhrs = classroom.sessions.fetchForAllClassroomMembers(classroom)
-        if jqxhrs.length > 0
-          @supermodel.trackRequests(jqxhrs)
+        Promise.all(classroom.sessions.fetchForAllClassroomMembers(classroom))
+        .then (results) =>
+          return if @destroyed
+          helper.calculateDots(@classrooms, @courses, @courseInstances)
+          @calculateQuestCompletion()
+          @render()
```

Comment:
Did you test this if one fails?

Reply:
Yes, it leaves the class in a loading state with all the course bubbles grey, hover popup says Loading..